source/extensions/filters/http/cache/config.cc

```diff
@@ -18,11 +18,12 @@ Http::FilterFactoryCb CacheFilterFactory::createFilterFactoryFromProtoTyped(
         fmt::format("Didn't find a registered implementation for type: '{}'", type));
   }
 
+  HttpCache* cache = &http_cache_factory->getCache(config, context);
```

Comment:
Is this definitely safe? Does this need to be a shared_ptr? I'm not sure of the lifetime of the various components here.

Reply:
I would tend to agree that it should be a shared_ptr, but that's a much broader change. This change is essentially a no-op except for efficiency and documentation - it's *already* required that the factory owns the pointer, or leaks it, since only a reference is returned, but before this change there's not even a comment to that effect.

Reply:
OK can you switch this to a TODO and we can move this to SingletonManager later?

/wait